nservicebus/architecture/publishing-from-web-applications.md

```diff
@@ -0,0 +1,90 @@
+
+Given these facts, conventional wisdom has suggested that when in the context of a web application, it is better to only send commands to a back-end service endpoint, which can then publish a similar event.
+
+But, things change:
+
+* NServiceBus 4.0 introduced the ability to have transports besides MSMQ. Not all of the transports handle Publish/Subscribe in the same way as MSMQ - some even support it natively.
+* Cloud transports generally have an associated cost per interaction. This means that requiring an unnecessary send before publish costs extra money to operate.
+* In NServiceBus 4.0 and 5.0, the process of creating a send-only endpoint (which is still capable of publishing messages, but does not have an input queue or process incoming messages) was made much easier.
+
+## Publish/Subscribe Mechanics
+
+First it's important to understand how publish/subscribe actually works, consisting of the subscription phase, and then the actual publish operation.
+
+### Subscription
+
+An endpoint must register its interest in a message by subscribing to a specific message type. How this occurs is dependent upon the message transport being used.
+
+Some transports (Azure Service Bus, RabbitMQ) support publish/subscribe natively. This means that when an endpoint wants to subcribe to an event, it contacts the broker directly, and the broker keeps track of subscribers for each event.
+
+For transports that lack native pub/sub capabilities (MSMQ, SQL, Azure Storage Queues) NServiceBus will provide you with similar semantics by using storage-driven publishing with message-driven subscriptions. This means that each endpoint is responsible for maintaining its own subscription storage, usually in a database. When an endpoint wants to subscribe to an event, it sends a subscription request message to the owner endpoint, which will update its own subscription storage.
+
+### Publishing
+
+The publishing process also differs based on the message transport.
+
+A transport that supports native publish/subscribe (Azure Service Bus, RabbitMQ) will send the event message directly to the broker, which will distribute the message to all registered subscribers.
+
+A transport that uses storage-driven publishing will act differently. When it needs to publish, it will first consult its subscription storage to determine the interested parties. It will then send an independent copy of the event message to each subscriber.
+
+While it is common to think of an event being published *from* a specific location, this doesn't always reflect the reality of the messaging mechanics. For native pub/sub brokers, there is no *from* concept. You simply publish an event and the broker figures out the rest.
+
+In the case of storage-driven transports, however, the "publish from" location actually relates to the input queue that is set up to receive subscription requests.
+
+## Load Balancing
+

```

Comment:
That really didn't work. The section needs to exist so that it leads into the Storage-driven topology section next. Did my best to indicate that hey, this isn't really any different than service endpoints.